### PR TITLE
chore: add PR template

### PR DESCRIPTION
**Pull request by bap2pecs**
## Summary

we need a template to remind us to not forget to test with FP. more context: https://github.com/babylonlabs-io/finality-gadget/pull/35#issuecomment-2452993076

follow follow https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/creating-a-pull-request-template-for-your-repository to create this template

## Test Plan

will until next PR

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+!!! Please test the code with https://github.com/babylonlabs-io/finality-provider
+
+Steps to test:
+1. `cd finality-provider`
+2. `git checkout base/consumer-chain-support`
+3. modify the `go.mod` to use the tip commit hash of the branch for this PR
+4. `go mod tidy`
+5. run `make test-e2e-op`
```
Comment (SebastianElvis) [inline .github/pull_request_template.md:8]: Can these be enforced in CI instead?

Comment (bap2pecs) [reply .github/pull_request_template.md:8]: good point. we can do that. we just need to write a script to automate this step:

```
3. modify the `go.mod` to use the tip commit hash of the branch for this PR
```

Comment (bap2pecs) [reply .github/pull_request_template.md:8]: we actually have a task https://github.com/babylonlabs-io/finality-gadget/issues/2

Comment (SebastianElvis) [reply .github/pull_request_template.md:8]: Hmm ok. Why not using this branch in `go.mod` directly?

Comment (SebastianElvis) [reply .github/pull_request_template.md:8]: By `go.mod` which repository are you referring to?

Comment (bap2pecs) [reply .github/pull_request_template.md:8]: `finality-provider`. the 2-4 steps below happen in the FP dir but it's triggered by the CI in this repo:

1. `cd finality-provider`
2. `git checkout base/consumer-chain-support`
3. modify the `go.mod` to use the tip commit hash of the branch for this PR
4. `go mod tidy`
5. run `make test-e2e-op`

Comment (SebastianElvis) [reply .github/pull_request_template.md:8]: Hmm I think now I get it. all good then 👍

**Review by SebastianElvis — APPROVED** [review]

**Pull request merged**